src/main/kotlin/org/cascadebot/cascadebot/commands/moderation/TempSlowmodeCommand.kt

```diff
@@ -0,0 +1,97 @@
+package org.cascadebot.cascadebot.commands.moderation
+
+import net.dv8tion.jda.api.Permission
+import net.dv8tion.jda.api.entities.Member
+import net.dv8tion.jda.api.entities.TextChannel
+import net.dv8tion.jda.api.exceptions.PermissionException
+import org.cascadebot.cascadebot.commandmeta.CommandContext
+import org.cascadebot.cascadebot.commandmeta.MainCommand
+import org.cascadebot.cascadebot.commandmeta.Module
+import org.cascadebot.cascadebot.data.managers.ScheduledActionManager
+import org.cascadebot.cascadebot.permissions.CascadePermission
+import org.cascadebot.cascadebot.scheduler.ActionType
+import org.cascadebot.cascadebot.scheduler.ScheduledAction
+import org.cascadebot.cascadebot.utils.DiscordUtils
+import org.cascadebot.cascadebot.utils.FormatUtils
+import org.cascadebot.cascadebot.utils.ParserUtils
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+
+class TempSlowmodeCommand : MainCommand() {
+
+    override fun onCommand(sender: Member, context: CommandContext) {
+        // <interval> <duration>  [channel]
+        if (context.args.size < 2) {
+            context.uiMessaging.replyUsage()
+            return
+        }
+
+        val longInterval = ParserUtils.parseTextTime(context.getArg(0), false) // In milliseconds
+        if (longInterval / 1000 > 21600) {  // 6 hour maximum time imposed by API
+            context.typedMessaging.replyWarning(context.i18n("commands.slowmode.time_exceeded"))
+            return
+        }
+
+        val longDuration = ParserUtils.parseTextTime(context.getArg(1), false)
+        if (longDuration <= 0) {
+            context.typedMessaging.replyDanger(context.i18n("responses.invalid_duration"))
+            return
+        }
+
+        var channel: TextChannel = context.channel
```

Comment:
You can change this to `TextChannel?` to allow nulls.

Reply:
> You can change this to `TextChannel?` to allow nulls.

Not advisable as it affects later code

Reply:
Ok then maybe he needs to have the temp channel then

Reply:
I don't like it as its an extra variable that could not exist, but megh.